src/Grpc.Net.Client/Internal/GrpcCallSerializationContext.cs

```diff
@@ -198,7 +200,7 @@ private IBufferWriter<byte> ResolveBufferWriter()
             {
                 if (_buffer == null)
                 {
-                    _buffer = ArrayPool<byte>.Shared.Rent(GrpcProtocolConstants.HeaderSize + _payloadLength.GetValueOrDefault());
+                    _buffer = ArrayPool<byte>.Shared.Rent(GrpcProtocolConstants.HeaderSize + _payloadLength.Value);
```

Comment:
Here it looks like the [JIT can't follow](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABABgAJiBGAOgCEBXAMyZilwG4BYAKF4EsAdhjaDsAG3K4YEmABNKAJnIBxKAAcwAYQniAym34T+AL2wZ+EQVqsiEGXgG9e5V+XVR+AN3MxykQVwMciFggAkZOTY9Uz8AXnIqRW4eN3dPHxEQ4QB+cgB9dWwAT3EIbDkAGRhBAHMMAAsUtI9vX3IIYAArGDBg/Mh8DxhcXEtBAAUoCC9+KKhmt1bMv2BikQBtAF0C4GZWBZcljPaAQSgoEsYWNgB1TxEoAB41kQA+Xf27h7ZF1yPXMt2sAIBBJAARfiwPoGTzGMwWKx6BjqdTQEQKOIfAYQIawUbjKYzOZschxBKCBjiSQAMhpBSKpXKVRq9Qa5AAhBSqeI/uQAek2llOj0+uQAEojMFeGDXA73fiPAAUAEoBc5Umk3PwmOQlZDoRhYUZxKZzONkaj0fI1ZqteQNfb7Tq9fk9jcoGTudTbU6tY6/U63V9PQlzpdihNQeIXusYG8aHoGthYHIaJLhEqIhVorFyABqBklMoVap1Ro0ABqEgYMBVfL9AF9eAK/cQAOzkRr8TittzNu1amDiaQhXVK4MehWPL3kSk+vuuAOB8gAelX5AAkoJFSa8+6DuQAO4/T1HxXsmAIbBi8Ssxpj8jYHz8cTYYB3miLrXr8jhq4htOpJyBAIyCAA5MEugQEe5AmGwEDZLuEjkHe5YNF+g5+pO8qnmSRZMqW97sh8pDfn6eSCDAsH/sUcrfIqbCxu8E6MiWLLoVWNZ1uRTogHO1F/hcAFTqezHxqqDb9i2WFuB2nyiYxhyDgOaQCkCwrdL0wSSrg0qyoBp5UKq6rfi6+pQtpxrwuaSIomiUAYr6TrLoG5k4aS5Jzjyzkrq5K6uB5oZCRGUZguJCZJim8jpjUGBZpEubwQWBHsWWbI0CoMAYNW4i1gA8lA4IwEw2BUvFKr1rxrgDtVlCdt2vayTV37DqO7kHgxM5efO4i+f6dU+J6bHMulD4JIUxajcRmXZblBVFSVZXiBVUn2nVv7bshprJZ1Z54eeD5XjewRoWyj7Ptgr7vp+G0brR9FQEBnogWBkFPtSMFwQhSEWChZ0VnVQXPfhI1Eeh5CkXVaSUYJD2GUpEVKmDHFsv1AXkPxVE0cJdEI48SNVc15C1cT8nA6efIDo2QA===) -- shown by `L00f5: call 0x00007ffc9e4254e8` which is the exception being thrown.

```suggestion
                    _buffer = ArrayPool<byte>.Shared.Rent(GrpcProtocolConstants.HeaderSize + _payloadLength.GetValueOrDefault());
```